sphinx/themes/basic/static/doctools.js

```diff
@@ -291,27 +288,52 @@ var Documentation = {
   },
 
   initOnKeyListeners: function() {
+    // only install a listener if it is really needed
+    if (!DOCUMENTATION_OPTIONS.NAVIGATION_WITH_KEYS &&
+        !DOCUMENTATION_OPTIONS.SEARCH_WITH_KEYS     &&
+        !DOCUMENTATION_OPTIONS.REMOVE_HIGHLIGHT_WITH_KEYS)
+        return;
     $(document).keydown(function(event) {
       var activeElementType = document.activeElement.tagName;
       // don't navigate when in search box, textarea, dropdown or button
       if (activeElementType !== 'TEXTAREA' && activeElementType !== 'INPUT' && activeElementType !== 'SELECT'
-          && activeElementType !== 'BUTTON' && !event.altKey && !event.ctrlKey && !event.metaKey
-          && !event.shiftKey) {
-        switch (event.keyCode) {
-          case 37: // left
-            var prevHref = $('link[rel="prev"]').prop('href');
-            if (prevHref) {
-              window.location.href = prevHref;
+          && activeElementType !== 'BUTTON') {
+        if (!event.altKey && !event.ctrlKey && !event.metaKey && !event.shiftKey) {
```

Comment:
It would probably be clearer if `altKey`, `ctrlKey` and `metaKey` stayed in the outer `if` and this only contained `shiftKey`, which this is all about, right?

Reply:
Reworked.